internal/jwks/cache_ttl.go

```diff
@@ -0,0 +1,133 @@
+package jwks
+
+import (
+	"context"
+	"sync"
+	"time"
+)
+
+type item struct {
+	sync.RWMutex
+	data       *JWK
+	expiration *time.Time
+}
+
+func (i *item) touch(d time.Duration) {
+	i.Lock()
+	exp := time.Now().Add(d)
+	i.expiration = &exp
+	i.Unlock()
+}
+
+func (i *item) expired() bool {
+	i.RLock()
+	res := true
+	if i.expiration != nil {
+		res = i.expiration.Before(time.Now())
+	}
+	i.RUnlock()
+	return res
+}
+
+// TTLCache is a TTL bases in-memory cache.
+type TTLCache struct {
+	mu    sync.RWMutex
+	ttl   time.Duration
+	stop  chan struct{}
+	items map[string]*item
+}
+
+// NewTTLCache returns a new instance of ttl cache.
+func NewTTLCache(ttl time.Duration) *TTLCache {
+	cache := &TTLCache{
+		ttl:   ttl,
+		stop:  make(chan struct{}),
+		items: make(map[string]*item),
+	}
+	cache.run()
+	return cache
+}
+
+func (tc *TTLCache) cleanup() {
+	tc.mu.Lock()
+	for key, item := range tc.items {
+		if item.expired() {
+			delete(tc.items, key)
+		}
+	}
+	tc.mu.Unlock()
+}
+
+func (tc *TTLCache) run() {
+	d := tc.ttl
+	if d < time.Second {
+		d = time.Second
+	}
+
+	ticker := time.Tick(d)
+	go func() {
+		for {
+			select {
+			case <-ticker:
+				tc.cleanup()
+			case <-tc.stop:
```

Comment:
Is there a way to stop cache at the moment?

Reply:
https://github.com/centrifugal/centrifugo/pull/410/files#diff-f8bd2f65ee773d1c71e5c047d159b86f1729df216e4f3f30fefb59de2368eb74R136